modules/generator/storage/config_util.go

```diff
@@ -36,6 +36,15 @@ func generateTenantRemoteWriteConfigs(originalCfgs []prometheus_config.RemoteWri
 			cloneCfg.Headers[user.OrgIDHeaderName] = tenant
 		}
 
+		if removeOrgIDHeader {
```

Comment:
can we just not add the header above instead of adding it and removing it here?

Reply:
@joe-elliott  Thanks for the review.
 Changes has been committed to skip addition of OrgIDHeader if config to add id header is set to false, please review the same https://github.com/grafana/tempo/pull/2974/commits/2ac2622c01979c901674f3e7610e3ed5af5819db